src/components/Results/ResultCard.jsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import { Card, Text, Tooltip } from "pi-ui";
+import CopyIcon from "src/assets/copy_icon.svg";
+import FileIcon from "src/assets/file_icon.svg";
+import HashIcon from "src/assets/hash_icon.svg";
+import DownloadIcon from "src/assets/download_icon.svg";
+import styles from "./Results.module.css";
+
+const ResultCard = ({
```

Comment:
Basically, this component is too big and difficult to maintain and should be broken down into smaller components. You can create Name, Digest, ChainInfo and DownloadProof components, for example.